Submitted By: Mario Fetka (mario dot fetka at gmail dot com)
Date: 2010-11-07
Initial Package Version: 2.13.2
Origin: me
Upstream Status: unknown
Description: correct the linkorder in the make files
 
diff -Naur cc65-2.13.2.orig/src/ar65/make/gcc.mak cc65-2.13.2/src/ar65/make/gcc.mak
--- cc65-2.13.2.orig/src/ar65/make/gcc.mak	2010-11-07 05:47:25.000000000 +0000
+++ cc65-2.13.2/src/ar65/make/gcc.mak	2010-11-07 05:51:00.000000000 +0000
@@ -51,7 +51,7 @@
 endif
 
 $(EXE):	$(OBJS) $(LIBS)
-	$(CC) $^ $(LDFLAGS) -o $@
+	$(CC) $(LDFLAGS) $^ -o $@
 	@if [ $(OS2_SHELL) ] ;	then $(EBIND) $(EXE) ; fi
 
 clean:
diff -Naur cc65-2.13.2.orig/src/ca65/make/gcc.mak cc65-2.13.2/src/ca65/make/gcc.mak
--- cc65-2.13.2.orig/src/ca65/make/gcc.mak	2010-11-07 05:47:25.000000000 +0000
+++ cc65-2.13.2/src/ca65/make/gcc.mak	2010-11-07 05:50:17.000000000 +0000
@@ -93,7 +93,7 @@
 endif
 
 $(EXE):	$(INCS) $(OBJS) $(LIBS)
-	$(CC) $(OBJS) $(LIBS) $(LDFLAGS) -o $@
+	$(CC) $(LDFLAGS) $(OBJS) $(LIBS) -o $@
 	@if [ $(OS2_SHELL) ] ;	then $(EBIND) $(EXE) ; fi
 
 inc:	$(INCS)
diff -Naur cc65-2.13.2.orig/src/cc65/make/gcc.mak cc65-2.13.2/src/cc65/make/gcc.mak
--- cc65-2.13.2.orig/src/cc65/make/gcc.mak	2010-11-07 05:47:24.000000000 +0000
+++ cc65-2.13.2/src/cc65/make/gcc.mak	2010-11-07 05:48:00.000000000 +0000
@@ -127,7 +127,7 @@
 endif
 
 $(EXE):	$(OBJS) $(LIBS)
-	$(CC) $(OBJS) $(LIBS) $(LDFLAGS) -o $@
+	$(CC) $(LDFLAGS) $(OBJS) $(LIBS) -lm -o $@
 	@if [ $(OS2_SHELL) ] ;	then $(EBIND) $(EXE) ; fi
 
 .PHONY:	svnversion
diff -Naur cc65-2.13.2.orig/src/cl65/make/gcc.mak cc65-2.13.2/src/cl65/make/gcc.mak
--- cc65-2.13.2.orig/src/cl65/make/gcc.mak	2010-11-07 05:47:25.000000000 +0000
+++ cc65-2.13.2/src/cl65/make/gcc.mak	2010-11-07 05:51:15.000000000 +0000
@@ -44,7 +44,7 @@
 endif
 
 $(EXE):	$(OBJS) $(LIBS)
-	$(CC) $^ $(LDFLAGS) -o $@
+	$(CC) $(LDFLAGS) $^ -o $@
 	@if [ $(OS2_SHELL) ] ;	then $(EBIND) $(EXE) ; fi
 
 clean:
diff -Naur cc65-2.13.2.orig/src/co65/make/gcc.mak cc65-2.13.2/src/co65/make/gcc.mak
--- cc65-2.13.2.orig/src/co65/make/gcc.mak	2010-11-07 05:47:25.000000000 +0000
+++ cc65-2.13.2/src/co65/make/gcc.mak	2010-11-07 05:50:34.000000000 +0000
@@ -44,7 +44,7 @@
 endif
 
 $(EXE):	$(OBJS) $(LIBS)
-	$(CC) $^ $(LDFLAGS) -o $@
+	$(CC) $(LDFLAGS) $^ -o $@
 	@if [ $(OS2_SHELL) ] ;	then $(EBIND) $(EXE) ; fi
 
 clean:
diff -Naur cc65-2.13.2.orig/src/da65/make/gcc.mak cc65-2.13.2/src/da65/make/gcc.mak
--- cc65-2.13.2.orig/src/da65/make/gcc.mak	2010-11-07 05:47:25.000000000 +0000
+++ cc65-2.13.2/src/da65/make/gcc.mak	2010-11-07 05:49:57.000000000 +0000
@@ -58,7 +58,7 @@
 endif
 
 $(EXE):	$(OBJS) $(LIBS)
-	$(CC) $^ $(LDFLAGS) -o $@
+	$(CC) $(LDFLAGS) $^ -o $@
 	@if [ $(OS2_SHELL) ] ;	then $(EBIND) $(EXE) ; fi
 
 clean:
diff -Naur cc65-2.13.2.orig/src/grc/make/gcc.mak cc65-2.13.2/src/grc/make/gcc.mak
--- cc65-2.13.2.orig/src/grc/make/gcc.mak	2010-11-07 05:47:25.000000000 +0000
+++ cc65-2.13.2/src/grc/make/gcc.mak	2010-11-07 05:50:47.000000000 +0000
@@ -38,7 +38,7 @@
 endif
 
 $(EXE):	$(OBJS) $(LIBS)
-	$(CC) $^ $(LDFLAGS) -o $@
+	$(CC) $(LDFLAGS) $^ -o $@
 	@if [ $(OS2_SHELL) ] ;	then $(EBIND) $(EXE) ; fi
 
 clean:
diff -Naur cc65-2.13.2.orig/src/ld65/make/gcc.mak cc65-2.13.2/src/ld65/make/gcc.mak
--- cc65-2.13.2.orig/src/ld65/make/gcc.mak	2010-11-07 05:47:25.000000000 +0000
+++ cc65-2.13.2/src/ld65/make/gcc.mak	2010-11-07 05:49:43.000000000 +0000
@@ -100,7 +100,7 @@
 endif
 
 $(EXE):	$(INCS) $(OBJS) $(LIBS)
-	$(CC) $(OBJS) $(LIBS) $(LDFLAGS) -o $@
+	$(CC) $(LDFLAGS) $(OBJS) $(LIBS) -o $@
 	@if [ $(OS2_SHELL) ] ;	then $(EBIND) $(EXE) ; fi
 
 inc:	$(INCS)
diff -Naur cc65-2.13.2.orig/src/od65/make/gcc.mak cc65-2.13.2/src/od65/make/gcc.mak
--- cc65-2.13.2.orig/src/od65/make/gcc.mak	2010-11-07 05:47:25.000000000 +0000
+++ cc65-2.13.2/src/od65/make/gcc.mak	2010-11-07 05:49:09.000000000 +0000
@@ -44,7 +44,7 @@
 endif
 
 $(EXE):	$(OBJS) $(LIBS)
-	$(CC) $^ $(LDFLAGS) -o $@
+	$(CC) $(LDFLAGS) $^ -o $@
 	@if [ $(OS2_SHELL) ] ;	then $(EBIND) $(EXE) ; fi
 
 clean:
diff -Naur cc65-2.13.2.orig/src/sim65/make/gcc.mak cc65-2.13.2/src/sim65/make/gcc.mak
--- cc65-2.13.2.orig/src/sim65/make/gcc.mak	2010-11-07 05:47:24.000000000 +0000
+++ cc65-2.13.2/src/sim65/make/gcc.mak	2010-11-07 05:48:47.000000000 +0000
@@ -53,7 +53,7 @@
 endif
 
 $(EXE):	$(OBJS) $(LIBS)
-	$(CC) $^ $(LDFLAGS) -o $@
+	$(CC) $(LDFLAGS) $^ -o $@
 	@if [ $(OS2_SHELL) ] ;	then $(EBIND) $(EXE) ; fi
 
 .PHONY:	chips
